File: Source/Alternet.UI.Pal/AuiPaneInfo.cpp
#include "AuiPaneInfo.h"

namespace Alternet::UI
{
	AuiPaneInfo::AuiPaneInfo()
	{

	}

	AuiPaneInfo::~AuiPaneInfo()
	{
	}

	wxAuiPaneInfo& AuiPaneInfo::PaneInfo(void* paneInfo)
	{
		AuiPaneInfo* paneInfoContainer = (AuiPaneInfo*)paneInfo;
		return paneInfoContainer->_paneInfo;
	}

	void AuiPaneInfo::Delete(void* handle) 
	{
		delete (AuiPaneInfo*)handle;
	}
	
	void* AuiPaneInfo::CreateAuiPaneInfo()
	{
		return new AuiPaneInfo();
	}

	void AuiPaneInfo::AuiPaneInfo::SafeSet(void* handle, void* source)
	{
		PaneInfo(handle).SafeSet(PaneInfo(source));
	}

	bool AuiPaneInfo::IsOk(void* handle)
	{
		return PaneInfo(handle).IsOk();
	}

	bool AuiPaneInfo::IsFixed(void* handle)
	{
		return PaneInfo(handle).IsFixed();
	}

	bool AuiPaneInfo::IsResizable(void* handle)
	{
		return PaneInfo(handle).IsResizable();
	}

	bool AuiPaneInfo::IsShown(void* handle)
	{
		return PaneInfo(handle).IsShown();
	}

	bool AuiPaneInfo::IsFloating(void* handle)
	{
		return PaneInfo(handle).IsFloating();
	}

	bool AuiPaneInfo::IsDocked(void* handle)
	{
		return PaneInfo(handle).IsDocked();
	}

	Int32Size AuiPaneInfo::GetBestSize(void* handle)
	{
		return PaneInfo(handle).best_size;
	}

	Int32Size AuiPaneInfo::GetMinSize(void* handle)
	{
		return PaneInfo(handle).min_size;
	}

	Int32Size AuiPaneInfo::GetMaxSize(void* handle)
	{
		return PaneInfo(handle).max_size;
	}

	bool AuiPaneInfo::IsToolbar(void* handle)
	{
		return PaneInfo(handle).IsToolbar();
	}

	bool AuiPaneInfo::IsTopDockable(void* handle)
	{
		return PaneInfo(handle).IsTopDockable();
	}

	bool AuiPaneInfo::IsBottomDockable(void* handle)
	{
		return PaneInfo(handle).IsBottomDockable();
	}

	bool AuiPaneInfo::IsLeftDockable(void* handle)
	{
		return PaneInfo(handle).IsLeftDockable();
	}

	bool AuiPaneInfo::IsRightDockable(void* handle)
	{
		return PaneInfo(handle).IsRightDockable();
	}

	bool AuiPaneInfo::IsDockable(void* handle)
	{	
		return PaneInfo(handle).IsDockable();
	}

	bool AuiPaneInfo::IsFloatable(void* handle)
	{
		return PaneInfo(handle).IsFloatable();
	}

	bool AuiPaneInfo::IsMovable(void* handle)
	{
		return PaneInfo(handle).IsMovable();
	}

	bool AuiPaneInfo::IsDestroyOnClose(void* handle)
	{
		return PaneInfo(handle).IsDestroyOnClose();
	}

	bool AuiPaneInfo::IsMaximized(void* handle)
	{
		return PaneInfo(handle).IsMaximized();
	}

	bool AuiPaneInfo::HasCaption(void* handle)
	{
		return PaneInfo(handle).HasCaption();
	}

	bool AuiPaneInfo::HasGripper(void* handle)
	{
		return PaneInfo(handle).HasGripper();
	}
	
	bool AuiPaneInfo::HasBorder(void* handle)
	{
		return PaneInfo(handle).HasBorder();
	}
	
	bool AuiPaneInfo::HasCloseButton(void* handle)
	{
		return PaneInfo(handle).HasCloseButton();
	}
	
	bool AuiPaneInfo::HasMaximizeButton(void* handle)
	{
		return PaneInfo(handle).HasMaximizeButton();
	}
	
	bool AuiPaneInfo::HasMinimizeButton(void* handle)
	{
		return PaneInfo(handle).HasMinimizeButton();
	}
	
	bool AuiPaneInfo::HasPinButton(void* handle)
	{
		return PaneInfo(handle).HasPinButton();
	}
	
	bool AuiPaneInfo::HasGripperTop(void* handle)
	{
		return PaneInfo(handle).HasGripperTop();
	}
	
	bool AuiPaneInfo::IsValid(void* handle)
	{
		return PaneInfo(handle).IsValid();
	}

	bool AuiPaneInfo::HasFlag(void* handle, int flag)
	{
		return PaneInfo(handle).HasFlag(flag);
	}

	void AuiPaneInfo::Window(void* handle, void* window)
	{
		PaneInfo(handle).Window((wxWindow*)window);
	}
	
	void AuiPaneInfo::Name(void* handle, const string& value)
	{
		PaneInfo(handle).Name(wxStr(value));
	}
	
	void AuiPaneInfo::Caption(void* handle, const string& value)
	{
		PaneInfo(handle).Caption(wxStr(value));
	}
	
	void AuiPaneInfo::Image(void* handle, ImageSet* bitmap)
	{
		PaneInfo(handle).Icon(ImageSet::BitmapBundle(bitmap));
	}
	
	void AuiPaneInfo::Left(void* handle)
	{
		PaneInfo(handle).Left();
	}
	
	void AuiPaneInfo::Right(void* handle)
	{
		PaneInfo(handle).Right();
	}
	
	void AuiPaneInfo::Top(void* handle)
	{
		PaneInfo(handle).Top();
	}
	
	void AuiPaneInfo::Bottom(void* handle)
	{
		PaneInfo(handle).Bottom();
	}
	
	void AuiPaneInfo::Center(void* handle)
	{
		PaneInfo(handle).Center();
	}
	
	void AuiPaneInfo::Direction(void* handle, int direction)
	{
		PaneInfo(handle).Direction(direction);
	}
	
	void AuiPaneInfo::Layer(void* handle, int layer)
	{
		PaneInfo(handle).Layer(layer);
	}
	
	void AuiPaneInfo::Row(void* handle, int row)
	{
		PaneInfo(handle).Row(row);
	}
	
	void AuiPaneInfo::Position(void* handle, int pos)
	{
		PaneInfo(handle).Position(pos);
	}
	
	void AuiPaneInfo::BestSize(void* handle, int x, int y)
	{
		PaneInfo(handle).BestSize(x, y);
	}
	
	void AuiPaneInfo::MinSize(void* handle, int x, int y)
	{
		PaneInfo(handle).MinSize(x, y);
	}
	
	void AuiPaneInfo::MaxSize(void* handle, int x, int y)
	{
		PaneInfo(handle).MaxSize(x, y);
	}
	
	void AuiPaneInfo::FloatingPosition(void* handle, int x, int y)
	{
		PaneInfo(handle).FloatingPosition(x, y);
	}
	
	void AuiPaneInfo::FloatingSize(void* handle, int x, int y)
	{
		PaneInfo(handle).FloatingSize(x, y);
	}
	
	void AuiPaneInfo::Fixed(void* handle)
	{
		PaneInfo(handle).Fixed();

		if (PaneInfo(handle).IsFixed())
		{
			if (true) {}
		}
	}
	
	void AuiPaneInfo::Resizable(void* handle, bool resizable)
	{
		PaneInfo(handle).Resizable(resizable);
	}
	
	void AuiPaneInfo::Dock(void* handle)
	{
		PaneInfo(handle).Dock();
	}
	
	void AuiPaneInfo::Float(void* handle)
	{
		PaneInfo(handle).Float();
	}
	
	void AuiPaneInfo::Hide(void* handle)
	{
		PaneInfo(handle).Hide();
	}
	
	void AuiPaneInfo::Show(void* handle, bool show)
	{
		PaneInfo(handle).Show(show);
	}
	
	void AuiPaneInfo::CaptionVisible(void* handle, bool visible)
	{
		PaneInfo(handle).CaptionVisible(visible);
	}

	void AuiPaneInfo::Maximize(void* handle)
	{
		PaneInfo(handle).Maximize();
	}

	void AuiPaneInfo::Restore(void* handle)
	{
		PaneInfo(handle).Restore();
	}

	void AuiPaneInfo::PaneBorder(void* handle, bool visible)
	{
		PaneInfo(handle).PaneBorder(visible);
	}

	void AuiPaneInfo::Gripper(void* handle, bool visible)
	{
		PaneInfo(handle).Gripper(visible);
	}

	void AuiPaneInfo::GripperTop(void* handle, bool attop)
	{
		PaneInfo(handle).GripperTop(attop);
	}

	void AuiPaneInfo::CloseButton(void* handle, bool visible)
	{
		PaneInfo(handle).CloseButton(visible);
	}

	void AuiPaneInfo::MaximizeButton(void* handle, bool visible)
	{
		PaneInfo(handle).MaximizeButton(visible);
	}

	void AuiPaneInfo::MinimizeButton(void* handle, bool visible)
	{
		PaneInfo(handle).MinimizeButton(visible);
	}

	void AuiPaneInfo::PinButton(void* handle, bool visible)
	{
		PaneInfo(handle).PinButton(visible);
	}

	void AuiPaneInfo::DestroyOnClose(void* handle, bool b)
	{
		PaneInfo(handle).DestroyOnClose(b);
	}

	void AuiPaneInfo::TopDockable(void* handle, bool b)
	{
		PaneInfo(handle).TopDockable(b);
	}

	void AuiPaneInfo::BottomDockable(void* handle, bool b)
	{
		PaneInfo(handle).BottomDockable(b);
	}

	void AuiPaneInfo::LeftDockable(void* handle, bool b)
	{
		PaneInfo(handle).LeftDockable(b);
	}

	void AuiPaneInfo::RightDockable(void* handle, bool b)
	{
		PaneInfo(handle).RightDockable(b);
	}

	void AuiPaneInfo::Floatable(void* handle, bool b)
	{
		PaneInfo(handle).Floatable(b);
	}

	void AuiPaneInfo::Movable(void* handle, bool b)
	{
		PaneInfo(handle).Movable(b);
	}

	void AuiPaneInfo::DockFixed(void* handle, bool b)
	{
		PaneInfo(handle).DockFixed(b);
	}

	void AuiPaneInfo::Dockable(void* handle, bool b)
	{
		PaneInfo(handle).Dockable(b);
	}

	void AuiPaneInfo::DefaultPane(void* handle)
	{
		PaneInfo(handle).DefaultPane();
	}

	void AuiPaneInfo::CenterPane(void* handle)
	{
		PaneInfo(handle).CenterPane();
	}

	void AuiPaneInfo::ToolbarPane(void* handle)
	{
		PaneInfo(handle).ToolbarPane();
	}

	void AuiPaneInfo::SetFlag(void* handle, int flag, bool option_state)
	{
		PaneInfo(handle).SetFlag(flag, option_state);
	}

}

File: Source/NativeApi/NativeApi/Api/AuiManager/AuiPaneInfo.cs
#pragma warning disable
using ApiCommon;
using System;
using System.Collections.Generic;
using System.Text;
using Alternet.Drawing;

namespace NativeApi.Api
{
	//https://docs.wxwidgets.org/3.2/classwx_aui_pane_info.html
	public class AuiPaneInfo
	{
        // size that the layout engine will prefer
        public static SizeI GetBestSize(IntPtr handle) => default;

        // minimum size the pane window can tolerate
        public static SizeI GetMinSize(IntPtr handle) => default;

        // maximum size the pane window can tolerate
        public static SizeI GetMaxSize(IntPtr handle) => default;

        public static void Delete(IntPtr handle) { }
        public static IntPtr CreateAuiPaneInfo() => throw new Exception();

        public static void SafeSet(IntPtr handle, IntPtr source) =>
            throw new Exception();
        public static bool IsOk(IntPtr handle) => throw new Exception();
        public static bool IsFixed(IntPtr handle) => throw new Exception();
        public static bool IsResizable(IntPtr handle) => throw new Exception();
        public static bool IsShown(IntPtr handle) => throw new Exception();
        public static bool IsFloating(IntPtr handle) => throw new Exception();
        public static bool IsDocked(IntPtr handle) => throw new Exception();
        public static bool IsToolbar(IntPtr handle) => throw new Exception();
        public static bool IsTopDockable(IntPtr handle) => throw new Exception();
        public static bool IsBottomDockable(IntPtr handle) => throw new Exception();
        public static bool IsLeftDockable(IntPtr handle) => throw new Exception();
        public static bool IsRightDockable(IntPtr handle) => throw new Exception();
        public static bool IsDockable(IntPtr handle) => throw new Exception();
        public static bool IsFloatable(IntPtr handle) => throw new Exception();
        public static bool IsMovable(IntPtr handle) => throw new Exception();
        public static bool IsDestroyOnClose(IntPtr handle) => throw new Exception();
        public static bool IsMaximized(IntPtr handle) => throw new Exception();
        public static bool HasCaption(IntPtr handle) => throw new Exception();
        public static bool HasGripper(IntPtr handle) => throw new Exception();
        public static bool HasBorder(IntPtr handle) => throw new Exception();
        public static bool HasCloseButton(IntPtr handle) => throw new Exception();
        public static bool HasMaximizeButton(IntPtr handle) => throw new Exception();
        public static bool HasMinimizeButton(IntPtr handle) => throw new Exception();
        public static bool HasPinButton(IntPtr handle) => throw new Exception();
        public static bool HasGripperTop(IntPtr handle) => throw new Exception();

        public static void Window(IntPtr handle, IntPtr window) => throw new Exception();
        public static void Name(IntPtr handle, string value) => throw new Exception();
        public static void Caption(IntPtr handle, string value) => throw new Exception();
        public static void Image(IntPtr handle, ImageSet? bitmap) =>
            throw new Exception();
        public static void Left(IntPtr handle) => throw new Exception();
        public static void Right(IntPtr handle) => throw new Exception();
        public static void Top(IntPtr handle) => throw new Exception();
        public static void Bottom(IntPtr handle) => throw new Exception();
        public static void Center(IntPtr handle) => throw new Exception();
        public static void Direction(IntPtr handle, int direction) =>
            throw new Exception();
        public static void Layer(IntPtr handle, int layer) => throw new Exception();
        public static void Row(IntPtr handle, int row) => throw new Exception();
        public static void Position(IntPtr handle, int pos) => throw new Exception();
        public static void BestSize(IntPtr handle, int x, int y) =>
            throw new Exception();
        public static void MinSize(IntPtr handle, int x, int y) => throw new Exception();
        public static void MaxSize(IntPtr handle, int x, int y) => throw new Exception();
        public static void FloatingPosition(IntPtr handle, int x, int y) =>
            throw new Exception();
        public static void FloatingSize(IntPtr handle, int x, int y) =>
            throw new Exception();
        public static void Fixed(IntPtr handle) => throw new Exception();
        public static void Resizable(IntPtr handle, bool resizable = true) =>
            throw new Exception();
        public static void Dock(IntPtr handle) => throw new Exception();
        public static void Float(IntPtr handle) => throw new Exception();
        public static void Hide(IntPtr handle) => throw new Exception();
        public static void Show(IntPtr handle, bool show = true) =>
            throw new Exception();
        public static void CaptionVisible(IntPtr handle, bool visible = true)
            => throw new Exception();
        public static void Maximize(IntPtr handle) => throw new Exception();
        public static void Restore(IntPtr handle) => throw new Exception();
        public static void PaneBorder(IntPtr handle, bool visible = true) =>
            throw new Exception();
        
        public static void Gripper(IntPtr handle, bool visible = true) =>
            throw new Exception();
        public static void GripperTop(IntPtr handle, bool attop = true) =>
            throw new Exception();
        public static void CloseButton(IntPtr handle, bool visible = true) =>
            throw new Exception();
        public static void MaximizeButton(IntPtr handle, bool visible = true) =>
            throw new Exception();
        public static void MinimizeButton(IntPtr handle, bool visible = true) =>
            throw new Exception();
        public static void PinButton(IntPtr handle, bool visible = true) =>
            throw new Exception();
        public static void DestroyOnClose(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void TopDockable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void BottomDockable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void LeftDockable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void RightDockable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void Floatable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void Movable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void DockFixed(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static void Dockable(IntPtr handle, bool b = true) =>
            throw new Exception();
        public static bool IsValid(IntPtr handle) => throw new Exception();

        public static void DefaultPane(IntPtr handle) => throw new Exception();
        public static void CenterPane(IntPtr handle) => throw new Exception();
        public static void ToolbarPane(IntPtr handle) => throw new Exception();
        public static void SetFlag(IntPtr handle, int flag, bool option_state) =>
            throw new Exception();
        public static bool HasFlag(IntPtr handle, int flag) => throw new Exception();


    }
}

/*
 
// name of the pane
public static string GetName() => default;
 	
// caption displayed on the window 
public static string GetCaption() => default;

// icon of the pane, may be invalid or wxBitmap 
public static IntPtr GetIcon() => default;
	
// window that is in this pane 
public static IntPtr GetWindow() => default;
 	
// floating frame window that holds the pane, wxFrame* 
public static IntPtr 	GetFrame() => default;

// a combination of wxPaneState values 
uint GetState() => default;

// dock direction (top, bottom, left, right, center) 
int GetDockDirection() => default;
 
int GetDockLayer() => default;
 	layer number (0 = innermost layer)
 
int 	dock_row() => default;
 	row number on the docking bar (0 = first row)
 
int 	dock_pos() => default;
 	position inside the row (0 = first position)
 
wxPoint 	floating_pos() => default;
 	position while floating
 
wxSize 	floating_size() => default;
 	size while floating
 
int 	dock_proportion() => default;
 	proportion while docked
 
wxAuiPaneButtonArray 	buttons() => default;
 	buttons on the pane

// current rectangle (populated by wxAUI)  
wxRect 	rect() => default;
 	
 
 */